--- original
+++ modified
@@ -28,14 +28,23 @@
 //#define SVT_LOG_QUIET
 #ifndef SVT_LOG_QUIET
 
+void __svt_av1_rust_log_callback(SvtLogLevel level, const char* tag, const char* message);
+
+#define SVT_RUST_LOG(level, tag, format, ...) \
+    do {                                      \
+        char buffer[256];                     \
+        snprintf(buffer, 256, format, ##__VA_ARGS__); \
+        __svt_av1_rust_log_callback(level, tag, buffer); \
+    } while (0)
+
 //SVT_LOG will not output the prefix. you can contorl the output style.
-#define SVT_LOG(format, ...) svt_log(SVT_LOG_ALL, NULL, format, ##__VA_ARGS__)
+#define SVT_LOG(format, ...) SVT_RUST_LOG(SVT_LOG_ALL, NULL, format, ##__VA_ARGS__)
 
-#define SVT_DEBUG(format, ...) svt_log(SVT_LOG_DEBUG, LOG_TAG, format, ##__VA_ARGS__)
-#define SVT_INFO(format, ...) svt_log(SVT_LOG_INFO, LOG_TAG, format, ##__VA_ARGS__)
-#define SVT_WARN(format, ...) svt_log(SVT_LOG_WARN, LOG_TAG, format, ##__VA_ARGS__)
-#define SVT_ERROR(format, ...) svt_log(SVT_LOG_ERROR, LOG_TAG, format, ##__VA_ARGS__)
-#define SVT_FATAL(format, ...) svt_log(SVT_LOG_FATAL, LOG_TAG, format, ##__VA_ARGS__)
+#define SVT_DEBUG(format, ...) SVT_RUST_LOG(SVT_LOG_DEBUG, LOG_TAG, format, ##__VA_ARGS__)
+#define SVT_INFO(format, ...) SVT_RUST_LOG(SVT_LOG_INFO, LOG_TAG, format, ##__VA_ARGS__)
+#define SVT_WARN(format, ...) SVT_RUST_LOG(SVT_LOG_WARN, LOG_TAG, format, ##__VA_ARGS__)
+#define SVT_ERROR(format, ...) SVT_RUST_LOG(SVT_LOG_ERROR, LOG_TAG, format, ##__VA_ARGS__)
+#define SVT_FATAL(format, ...) SVT_RUST_LOG(SVT_LOG_FATAL, LOG_TAG, format, ##__VA_ARGS__)
 
 #else
 
